src/services/goals.js

```diff
@@ -194,141 +199,137 @@ async function cleanupObjectivesForGoal(goalId, currentObjectives) {
-  //           },
-  //           transaction,
-  //         }))),
-  //       );
-
-  //       // cleanup objective topics
-  //       await ObjectiveTopic.destroy({
-  //         where: {
-  //           id: {
-  //             [Op.notIn]: objectiveTopics.length ? objectiveTopics.map(([ot]) => ot.id) : [],
-  //           },
-  //           objectiveId: objective.id,
-  //         },
-  //       });
-
-  //       // resources
-  //       const objectiveResources = await Promise.all(
-  //         resources.filter(({ value }) => value).map(
-  //           ({ value }) => ObjectiveResource.findOrCreate({
-  //             where: {
-  //               userProvidedUrl: value,
-  //               objectiveId: objective.id,
-  //             },
-  //             transaction,
-  //           }),
-  //         ),
-  //       );
-
-  //       // cleanup objective resources
-  //       await ObjectiveResource.destroy({
-  //         where: {
-  //           id: {
-  //             [Op.notIn]: objectiveResources.length
-  //               ? objectiveResources.map(([or]) => or.id) : [],
-  //           },
-  //           objectiveId: objective.id,
-  //         },
-  //       });
+        // cleanup objective topics
+        await ObjectiveTopic.destroy({
+          where: {
+            id: {
+              [Op.notIn]: objectiveTopics.length ? objectiveTopics.map(([ot]) => ot.id) : [],
+            },
+            objectiveId: objective.id,
+          },
+        });
+
+        // resources
+        const objectiveResources = await Promise.all(
+          resources.filter(({ value }) => value).map(
+            ({ value }) => ObjectiveResource.findOrCreate({
+              where: {
+                userProvidedUrl: value,
+                objectiveId: objective.id,
+              },
+            }),
+          ),
+        );
+
+        // cleanup objective resources
+        await ObjectiveResource.destroy({
+          where: {
+            id: {
+              [Op.notIn]: objectiveResources.length
+                ? objectiveResources.map(([or]) => or.id) : [],
+            },
+            objectiveId: objective.id,
+          },
+        });
 
-  //       return {
-  //         ...objective.dataValues,
-  //         topics,
-  //         resources,
-  //       };
-  //     }),
-  //   );
+        return {
+          ...objective.dataValues,
+          topics,
+          resources,
+        };
+      }),
+    );
 
-  //   // this function deletes unused objectives
-  //   await cleanupObjectivesForGoal(goal.id, newObjectives);
+    // this function deletes unused objectives
+    await cleanupObjectivesForGoal(newGoal.id, newObjectives);
 
-  //   return goal.id;
-  // })));
+    return newGoal.id;
+  }));
 
-  // // we have to do this outside of the transaction otherwise
-  // // we get the old values
-  // return goalsByIdAndRecipient(goalIds, recipient);
+  // we have to do this outside of the transaction otherwise
+  // we get the old values
```

Comment:
This comment is no longer true because `goalsByIdAndRecipient` takes place in the same transaction as all the queries above